components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.websocket/src/main/java/org/wso2/carbon/event/output/adapter/websocket/WebsocketEventAdapter.java

```diff
@@ -121,26 +122,35 @@ public void connect() {
         try {
             session = client.connectToServer(new WebsocketClient(), clientEndpointConfig, new URI(socketServerUrl));
         } catch (DeploymentException e) {
-            throw new ConnectionUnavailableException("The adaptor " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
-                    socketServerUrl, e);
+            throw new ConnectionUnavailableException("The adapter " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
+                                                     socketServerUrl, e);
         } catch (IOException e) {
-            throw new ConnectionUnavailableException("The adaptor " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
-                    socketServerUrl, e);
+            throw new ConnectionUnavailableException("The adapter " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
+                                                     socketServerUrl, e);
         } catch (URISyntaxException e) {
-            throw new OutputEventAdapterRuntimeException("The adaptor " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
-                    socketServerUrl, e);
+            throw new OutputEventAdapterRuntimeException("The adapter " + eventAdapterConfiguration.getName() + " failed to connect to the websocket server " +
+                                                         socketServerUrl, e);
         }
     }
 
     @Override
     public void publish(Object message, Map<String, String> dynamicProperties) {
-
         try {
-            executorService.execute(new WebSocketSender(message.toString()));
+            if (session == null) {
+                throw new ConnectionUnavailableException("Connection was not available when trying to publish a message via the adapter: "
+                                                         + eventAdapterConfiguration.getName() + ", for tenant ID: " + tenantId
+                                                         + " as the session was null. \n Message: \n" + message);
+            } else if (!session.isOpen()) {
+                throw new ConnectionUnavailableException("Connection was not available when trying to publish a message via the adapter: "
+                                                         + eventAdapterConfiguration.getName() + ", for tenant ID: " + tenantId
+                                                         + " as the session was not open. Session ID:"
+                                                         + session.getId() + " \n Message: \n" + message);
```

Comment:
Can we shorten and reformat this error message as well, similar to what is suggested above?